### PR TITLE
custom changes for v8.27.0

### PR DESCRIPTION
**Pull request by aman-adpushup**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- contact email of the adapter’s maintainer
- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Comment by yashraj-02** [comment]
Don't we need to handle defaultTtl for coinzilla?
https://github.com/prebid/Prebid.js/compare/8.27.0...adpushup:Prebid.js:adpushup.js/v7.48.0-custom-changes
image

File: modules/rubiconBidAdapter.js
Comment (yashraj-02) [inline modules/rubiconBidAdapter.js:1]: we can eliminate changes in `rubiconBidAdapter.js` if we add `bidonmultiformat: true` in rubicon config inside  `data::apNetworks` in couchbase.

File: modules/ixBidAdapter.js [hunk 1]
```diff
@@ -1473,7 +1494,7 @@ function updateMissingSizes(validBidRequest, missingBannerSizes, imp) {
   } else {
     // New Ad Unit
     if (deepAccess(validBidRequest, 'mediaTypes.banner.sizes')) {
-      let sizeList = deepClone(validBidRequest.mediaTypes.banner.sizes);
+      let sizeList = deepClone(validBidRequest.mediaTypes.banner.sizes.filter(size => apIsValidSize(...size)));
```
Comment (yashraj-02) [inline modules/ixBidAdapter.js:1497]: Too much chaining. Better to store `sizes` in variable & then pass to `deepClone()`

File: modules/ixBidAdapter.js [hunk 2]
```diff
@@ -163,6 +163,27 @@ const MEDIA_TYPES = {
   Native: 4
 };
 
+const AP_VALID_SIZES = [
```
Comment (yashraj-02) [inline modules/ixBidAdapter.js:166]: Any idea why we have not mentioned all the IAB sizes in here?

`				[120, 600],
				[160, 600],
				[200, 200],
				[240, 400],
				[250, 250],
				[300, 50],
				[300, 100],
				[300, 250],
				[300, 600],
				[320, 50],
				[320, 100],
				[320, 480],
				[336, 280],
				[468, 60],
				[480, 320],
				[580, 400],
				[690, 90],
				[675, 90],
				[670, 90],
				[650, 90],
				[630, 90],
				[600, 90],
				[720, 300],
				[728, 90],
				[728, 250],
				[728, 280],
				[900, 90],
				[970, 90],
				[970, 250]`

File: modules/sharethroughBidAdapter.js
Comment (yashraj-02) [inline modules/sharethroughBidAdapter.js:1]: Please check with business team why do we need to disable outstream for sharethrough. If it was a past requirement check if it is still valid.

**Comment by yashraj-02** [comment]
1. We should have a Notion document for the Prebid upgrade. We should document all the required changes along with the reasons. In every Prebid upgrade, we should revisit those changes and check with the business teams whether they are still necessary or not. 
2. Additionally, during every Prebid upgrade, we should cross-check whether code changes are necessary or if there is any flag available in the latest Prebid version for our requirements, which could help us avoid making changes.

File: modules/eplanningBidAdapter.js [hunk 3]
```diff
@@ -36,76 +36,19 @@ export const spec = {
   },
 
   buildRequests: function(bidRequests, bidderRequest) {
```
Comment (yashraj-02) [inline modules/eplanningBidAdapter.js:38]: Here we have changed return type from object to array of object. Is it compatible?

File: modules/ixBidAdapter.js (hunk 1 above)
Comment (yashraj-02) [reply modules/ixBidAdapter.js:1497]: Additionally, don't we need to update the `currentSizeList` array on line 1492?

File: modules/eplanningBidAdapter.js (hunk 3 above)
Comment (aman-adpushup) [reply modules/eplanningBidAdapter.js:38]: I have checked showing no errors
Here is the screenshot
image
image

and same impletions is in rubicon bidadapter.

File: modules/sharethroughBidAdapter.js
Comment (aman-adpushup) [reply modules/sharethroughBidAdapter.js:1]: https://adpushup.slack.com/archives/C06RLQ395SQ/p1711532584994709

Comment (aman-adpushup) [reply modules/sharethroughBidAdapter.js:1]: Discussed on slack

File: modules/rubiconBidAdapter.js
Comment (aman-adpushup) [reply modules/rubiconBidAdapter.js:1]: I tried using bidonmultiformat:true.
but the bid was not showing for all ad units.
On the previous setup, the bid was showing on all adunits.

File: modules/ixBidAdapter.js (hunk 2 above)
Comment (aman-adpushup) [reply modules/ixBidAdapter.js:166]: https://adpushup.slack.com/archives/C06RLQ395SQ/p1711630186060059
Discussed on slack

File: modules/ixBidAdapter.js (hunk 1 above)
Comment (aman-adpushup) [reply modules/ixBidAdapter.js:1497]: Not needed

image

**Comment by aman-adpushup** [comment]
> Don't we need to handle defaultTtl for coinzilla? [prebid/Prebid.js@8.27.0...adpushup:Prebid.js:adpushup.js/v7.48.0-custom-changes](https://github.com/prebid/Prebid.js/compare/8.27.0...adpushup:Prebid.js:adpushup.js/v7.48.0-custom-changes) image

File: modules/ixBidAdapter.js (hunk 1 above)
Comment (aman-adpushup) [reply modules/ixBidAdapter.js]: currentSizeList updated when bidder request goes.

**Review by yashraj02 — APPROVED** [review]

**Pull request merged**